packages/components/src/components/pin-code/pin-code.tsx

```diff
@@ -0,0 +1,319 @@
+
+  /** Marks the Pin Code as required. */
+  @Prop() public required?: boolean = false;
+
+  /** The message styled depending on validation state. */
+  @Prop() public message?: string = '';
+
+  /** Pin Code type. */
+  @Prop() public type?: PinCodeType = 'number';
+
+  /** Sets the initial value of the Pin Code. */
+  @Prop({ mutable: true }) public value?: string = '';
+
+  /** Adapts the color depending on the theme. */
+  @Prop() public theme?: Theme = 'light';
+
+  /** Emitted when selected element changes. */
+  @Event({ bubbles: false }) public update: EventEmitter<PinCodeUpdateEvent>;
+
+  private form: HTMLFormElement;
+  private isWithinForm: boolean;
+  private hiddenInput: HTMLInputElement;
+  private pinCodeElements: HTMLInputElement[] = [];
+
+  public componentWillLoad(): void {
+    this.form = getClosestHTMLElement(this.host, 'form');
+    this.isWithinForm = !!this.form;
+    if (this.isWithinForm) {
+      this.hiddenInput = initHiddenInput(this.host, this.name, this.value, this.disabled, this.required);
+    }
+  }
+
+  public componentWillUpdate(): void {
+    if (this.isWithinForm) {
+      syncHiddenInput(this.hiddenInput, this.name, this.value, this.disabled, this.required);
+    }
+  }
+
+  public componentShouldUpdate(newVal: unknown, oldVal: unknown): boolean {
+    return hasPropValueChanged(newVal, oldVal);
+  }
+
+  public render(): JSX.Element {
+    validateProps(this, propTypes);
+    attachComponentCss(
+      this.host,
+      getComponentCss,
+      this.hideLabel,
+      this.state,
+      this.disabled,
+      this.loading,
+      this.isWithinForm,
+      this.length,
+      this.theme
+    );
+
+    const PrefixedTagNames = getPrefixedTagNames(this.host);
+    const currentInputId = 'current-input';
+
+    // reset array of input elements
+    this.pinCodeElements = [];
+
+    this.validateInitialValue();
+
+    return (
+      <Host>
+        <label class="label" htmlFor={currentInputId}>
+          {hasLabel(this.host, this.label) && (
+            <span id="label" class="label__text">
+              {this.label || <slot name="label" />}
+              {!isParentFieldsetRequired(this.host) && this.required && <Required />}
+            </span>
+          )}
+          {hasDescription(this.host, this.description) && (
+            <span id="description" class="label__text">
+              {this.description || <slot name="description" />}
+            </span>
+          )}
+        </label>
+        <div class="input-container" onKeyDown={this.onKeyDown} onPaste={this.onPaste} onInput={this.onInput}>
+          {this.loading && (
+            <PrefixedTagNames.pSpinner
+              class="spinner"
+              size="inherit"
+              theme={this.theme}
+              aria={{ 'aria-label': 'Loading state' }}
+            />
+          )}
+          {this.isWithinForm && <slot name={hiddenInputSlotName} />}
+          {...Array.from({ length: this.length }).map((_value, index) => (
```

Comment:
No need for array spreading I think.
```suggestion
          {Array.from(Array(this.length), (_, index) => (
```